enclave_build/src/main.rs

```diff
@@ -1,144 +1,127 @@
                 .help("Output file for EIF image")
-                .takes_value(true)
                 .required(true),
         )
         .arg(
-            Arg::with_name("signing-certificate")
+            Arg::new("signing-certificate")
                 .long("signing-certificate")
-                .help("Specify the path to the signing certificate")
-                .takes_value(true),
+                .help("Specify the path to the signing certificate"),
         )
         .arg(
-            Arg::with_name("private-key")
+            Arg::new("private-key")
                 .long("private-key")
-                .help("Specify the path to the private-key")
-                .takes_value(true),
+                .help("Specify the path to the private-key"),
         )
         .arg(
-            Arg::with_name("build")
+            Arg::new("build")
                 .short('b')
                 .long("build")
                 .help("Build image from Dockerfile")
-                .takes_value(true)
-                .required(false),
+                .conflicts_with("pull"),
         )
         .arg(
-            Arg::with_name("pull")
+            Arg::new("pull")
                 .short('p')
                 .long("pull")
                 .help("Pull the Docker image before generating EIF")
-                .required(false)
+                .action(ArgAction::SetTrue)
                 .conflicts_with("build"),
         )
         .arg(
-            Arg::with_name("image_name")
+            Arg::new("image_name")
                 .long("name")
-                .help("Name for enclave image")
-                .takes_value(true),
+                .help("Name for enclave image"),
         )
         .arg(
-            Arg::with_name("image_version")
+            Arg::new("image_version")
                 .long("version")
-                .help("Version of the enclave image")
-                .takes_value(true),
+                .help("Version of the enclave image"),
         )
         .arg(
-            Arg::with_name("metadata")
+            Arg::new("metadata")
                 .long("metadata")
-                .help("Path to JSON containing the custom metadata provided by the user.")
-                .takes_value(true),
+                .help("Path to JSON containing the custom metadata provided by the user"),
         )
         .get_matches();
 
-    let docker_image = matches.value_of("docker_image").unwrap();
-    let init_path = matches.value_of("init_path").unwrap();
-    let nsm_path = matches.value_of("nsm_path").unwrap();
-    let kernel_img_path = matches.value_of("kernel_img_path").unwrap();
-    let kernel_cfg_path = matches.value_of("kernel_cfg_path").unwrap();
-    let cmdline = matches.value_of("cmdline").unwrap();
-    let linuxkit_path = matches.value_of("linuxkit_path").unwrap();
-    let output = matches.value_of("output").unwrap();
+    let docker_image = matches.get_one::<String>("docker_image").unwrap();
```

Comment:
We should consider to refactor this to use the clap derive feature in the future.